recipe/meta.yaml

```diff
@@ -1,311 +1,309 @@
+    # In order to use JasPer, OpenJPEG must be disabled.
+    # see https://github.com/opencv/opencv/blob/4.10.0/doc/tutorials/introduction/config_reference/config_reference.markdown#png-jpeg-tiff-webp-support
     - jpeg 9e                  # [not win]
-    - openjpeg 2.3.0           # [not (win or arm64)]
-    - openblas-devel 0.3.20    # [x86_64 and not win]
-    - libpng 1.6.37
-    - libprotobuf 3.20.3       # [(linux and x86_64) or win]
-    - libtiff 4.1              # [not ((osx and arm64) or (linux and aarch64))]
-    - libtiff 4.2              # [(osx and arm64) or (linux and aarch64)]
-    - libwebp-base 1.2.0       # [win or (linux and x86_64)]
-    - numpy {{ numpy }}
-    - python
-    - qt-main >=5.15,<6         # [not (linux and (s390x or ppc64le)) and build_variant == 'normal']
-    - qt-webengine >=5.15.7,<6  # [not (linux and (s390x or ppc64le)) and build_variant == 'normal']
-    - qtwebkit =5               # [not (linux and (s390x or ppc64le)) and build_variant == 'normal']
-    - zlib 1.2.13
-  run:
-    - _openmp_mutex         # [linux]
-    - eigen >=3.3.7
-    - ffmpeg >=4.2.2,<4.3.0a0        # [not (linux and s390x)]
-    - freetype >=2.10.4,<3.0a0
-    - jasper          # [not s390x]
-    - glib >=2.69,<3.0a0                 # [not ppc64le]
-    - gst-plugins-base >=1.14.0,<1.15.0a0     # [not (win or ppc64le)]
-    - gstreamer >=1.14.0,<1.15.0a        # [not (win or ppc64le)]
-    - gst-plugins-base >=1.18.5,<1.19.0a # [win]
-    - gstreamer >=1.18.5,<1.19.0a # [win]
-    - harfbuzz >=4.3.0,<5.0a0             # [not win]
-    - hdf5 >=1.12.1,<1.12.2.0a0
-    - jpeg >=9e,<10a                 # [not win]
-    - openjpeg >=2.3.0,<3.0a0             # [not (win or arm64)]
-    - libopenblas >=0.3.20,<1.0a0          # [x86_64 and not win]
-    - libpng >=1.6.37,<1.7.0a0
-    - libtiff >=4.1.0,<5.0a0  # [not ((osx and arm64) or (linux and aarch64))]
-    - libtiff >=4.2.0,<5.0a0  # [(osx and arm64) or (linux and aarch64)]
-    - libwebp-base          # [win or (linux and x86_64)]
-    - numpy >={{ numpy }},<2
-    - python
-    - qt-main >=5.15,<6         # [not (linux and (s390x or ppc64le)) and build_variant == 'normal']
-    - qt-webengine >=5.15.7,<6  # [not (linux and (s390x or ppc64le)) and build_variant == 'normal']
-    - qtwebkit =5               # [not (linux and (s390x or ppc64le)) and build_variant == 'normal']
-    - zlib >=1.2.13,<1.3.0a0
+    - jasper 4.2.4             # [win or osx-arm64]
+    - openjpeg 2.3.0           # [not (win or osx-arm64)]
+    - libiconv                 # [unix]
+    - openblas-devel 0.3.21    # [x86_64 and not win]
+    - libpng 1.6.39
+    - libprotobuf
+    # to support protobuf >=22 and c++17
+    - libabseil 20240116.2
+    - libtiff 4.5
+    - libwebp-base 1.3.2        # [win or (linux and x86_64)]
+    - qt-main                   # [qt_version == 5 or (not (linux and (s390x or ppc64le)))]
```

Comment:
Overall, it would be good to use the cbc values as much as possible for pins `{{ .. }}` and pin every host dep. The qt dep could be pinned to 5.